### PR TITLE
[FIX] agreement: Issue #290

### LINKED ISSUE #290: [11.0][BUG] agreement: error on model's field selection in "filed_id"
In "agreement", "agreement.recital", "agreement.section", "agreement.clause", "agreement.appendix"; 
when selecting a field from the current model in "filed_id" (there is no problem when selecting a relational field).

LOG:
> 2019-03-27 09:40:14,855 1 ERROR vert odoo.http: Exception during JSON request handling. 
> odoo_1         | Traceback (most recent call last):
> odoo_1         |   File "/opt/odoo/odoo/http.py", line 654, in _handle_exception
> odoo_1         |     return super(JsonRequest, self)._handle_exception(exception)
> odoo_1         |   File "/opt/odoo/odoo/http.py", line 312, in _handle_exception
> odoo_1         |     raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
> odoo_1         |   File "/opt/odoo/odoo/tools/pycompat.py", line 87, in reraise
> odoo_1         |     raise value
> odoo_1         |   File "/opt/odoo/odoo/http.py", line 696, in dispatch
> odoo_1         |     result = self._call_function(**self.params)
> odoo_1         |   File "/opt/odoo/odoo/http.py", line 344, in _call_function
> odoo_1         |     return checked_call(self.db, *args, **kwargs)
> odoo_1         |   File "/opt/odoo/odoo/service/model.py", line 97, in wrapper
> odoo_1         |     return f(dbname, *args, **kwargs)
> odoo_1         |   File "/opt/odoo/odoo/http.py", line 337, in checked_call
> odoo_1         |     result = self.endpoint(*a, **kw)
> odoo_1         |   File "/opt/odoo/odoo/http.py", line 939, in __call__
> odoo_1         |     return self.method(*args, **kw)
> odoo_1         |   File "/opt/odoo/odoo/http.py", line 517, in response_wrap
> odoo_1         |     response = f(*args, **kw)
> odoo_1         |   File "/opt/odoo/addons/000/web/controllers/main.py", line 962, in call_kw
> odoo_1         |     return self._call_kw(model, method, args, kwargs)
> odoo_1         |   File "/opt/odoo/addons/000/web/controllers/main.py", line 954, in _call_kw
> odoo_1         |     return call_kw(request.env[model], method, args, kwargs)
> odoo_1         |   File "/opt/odoo/odoo/api.py", line 749, in call_kw
> odoo_1         |     return _call_kw_multi(method, model, args, kwargs)
> odoo_1         |   File "/opt/odoo/odoo/api.py", line 736, in _call_kw_multi
> odoo_1         |     result = method(recs, *args, **kwargs)
> odoo_1         |   File "/opt/odoo/odoo/models.py", line 5430, in onchange
> odoo_1         |     record._onchange_eval(name, field_onchange[name], result)
> odoo_1         |   File "/opt/odoo/odoo/models.py", line 5283, in _onchange_eval
> odoo_1         |     method_res = method(self)
> odoo_1         |   File "/opt/odoo/addons/090/agreement_legal/models/agreement_section.py", line 75, in onchange_sub_object_id
> odoo_1         |     )[0]
> odoo_1         |   File "/opt/odoo/odoo/models.py", line 5053, in __getitem__
> odoo_1         |     return self._browse((self._ids[key],), self.env)
> odoo_1         | IndexError: tuple index out of range
> odoo_1         | 2019-03-27 09:40:14,857 1 INFO vert werkzeug: 172.23.0.1 - - [27/Mar/2019 09:40:14] "POST /web/dataset/call_kw/agreement.section/onchange HTTP/1.1" 200 - 5 0.013 1090.319
> odoo_1         | 2019-03-27 09:40:15,037 wdb You can now launch your browser at http://localhost:1984/pm/session/749d712d-a8b8-489d-8bc7-cb5e95679176 at open_browser:792
>

### PR DESCRIPTION
**Pull request by max3903**
Fixes #290

### CODE REVIEW

**Comment by max3903** [comment]
@ygol Can you test this? This is working on my side.

File: agreement/models/agreement.py [hunk 1]
```diff
@@ -342,11 +342,15 @@ def onchange_sub_object_id(self):
 
     @api.onchange('sub_model_object_field_id', 'default_value')
```
Comment (ygol) [inline agreement/models/agreement.py]: I think "field_id" should be added in line 343:
 @api.onchange('sub_model_object_field_id', 'default_value', 'field_id')

Comment (ygol) [reply agreement/models/agreement.py]: it works but still when editing, the "sub_object_id" field is not reset and shows the previous value.

Comment (max3903) [reply agreement/models/agreement.py]: Got it. Should be fixed now.

File: agreement/models/agreement.py [hunk 2]
```diff
@@ -346,7 +346,7 @@ def onchange_copyvalue(self):
             self.sub_model_object_field_id = False
         if self.field_id and self.field_id.relation:
             self.sub_object_id = self.env['ir.model'].search(
-                    [('model', '=', self.field_id.relation)])[0]
+                [('model', '=', self.field_id.relation)])[0]
             if self.sub_model_object_field_id:
```
Comment (ygol) [inline agreement/models/agreement.py]:
```suggestion
        if self.sub_model_object_field_id:
            self.copyvalue = "${object.%s.%s or %s}" %\
                             (self.field_id.name,
                              self.sub_model_object_field_id.name,
                              self.default_value or '\'\'')

```
This works for me locally.  I don't seem to be able to get a working runbot instance (https://runbot.odoo-community.org/runbot/repo/github-com-oca-contract-110)

**Review by ygol — CHANGES_REQUESTED** [review]

**Review by ygol — APPROVED** [review]
LGTM

**Pull request merged**